src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1107,7 +1107,9 @@ var ReactCompositeComponent = {
     if (__DEV__) {
       var componentName = component && component.getName ?
         component.getName() : 'a component';
-      warning(publicComponentInstance != null,
+      warning(
+        publicComponentInstance != null ||
+        component._compositeType !== CompositeTypes.StatelessFunctional,
```

Comment:
Agree, sorry I missed this. Should we add a check for `typeof component._currentElement.type === 'function'`?